plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2448,7 +2468,58 @@ public ConnectorMergeTableHandle beginMerge(ConnectorSession session, ConnectorT
         DeltaLakeInsertTableHandle insertHandle = createInsertHandle(retryMode, handle, inputColumns);
 
         Map<String, DeletionVectorEntry> deletionVectors = loadDeletionVectors(session, handle);
-        return new DeltaLakeMergeTableHandle(handle, insertHandle, deletionVectors);
+        return new DeltaLakeMergeTableHandle(handle, insertHandle, deletionVectors, shallowCloneSourceTableLocation(session, handle));
+    }
+
+    private Optional<String> shallowCloneSourceTableLocation(ConnectorSession session, DeltaLakeTableHandle handle)
+    {
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+        String sourceTableName;
+        try {
+            // The clone commit is the first commit of the cloned table, so set the endVersion to 0
+            TransactionLogTail transactionLogTail = loadNewTail(fileSystem, handle.getLocation(), Optional.empty(), Optional.of(0L), DataSize.ofBytes(0));
+            List<Transaction> transactions = transactionLogTail.getTransactions();
+            if (transactions.isEmpty()) {
+                return Optional.empty();
+            }
+
+            Optional<CommitInfoEntry> cloneCommit = transactions.getFirst().transactionEntries().getEntries(fileSystem)
+                    .map(DeltaLakeTransactionLogEntry::getCommitInfo)
+                    .filter(Objects::nonNull)
+                    .filter(commitInfoEntry -> commitInfoEntry.operation().equals("CLONE"))
+                    .findFirst();
+            if (cloneCommit.isEmpty()) {
+                return Optional.empty();
+            }
+
+            // It's the cloned table
+            Map<String, String> operationParameters = cloneCommit.get().operationParameters();
+            if (!operationParameters.containsKey("source")) {
+                return Optional.empty();
+            }
+
+            sourceTableName = operationParameters.get("source");
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        if (sourceTableName == null || !sourceTableName.contains(".") || sourceTableName.split("\\.").length != 3) {
+            return Optional.empty();
+        }
+
+        String[] names = sourceTableName.split("\\.");
```

Comment:
What happens if the table name contains a dot? We can create such tables on Glue though it's unsupported on Hive Thrift metastore.

Reply:
I assume that the schema shouldn't contain dots.

Reply:
Revisit and thinking reject support the table name contain dots, as now, we not sure if catalogs or schemas may contain dots.

Reply:
updated